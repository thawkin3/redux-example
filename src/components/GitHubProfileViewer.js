import React from 'react';
import Form from './Form';
import CardList from './CardList';
import './GitHubProfileViewer.css';

const GitHubProfileViewer = () => (
    <div>
        <h3>Github Profile Viewer</h3>
        <Form />
        <CardList />
    </div>
);

GitHubProfileViewer.displayName = 'GitHubProfileViewer';

export default GitHubProfileViewer;
