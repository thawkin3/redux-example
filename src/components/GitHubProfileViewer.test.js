import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import reduxThunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import GitHubProfileViewer from './GitHubProfileViewer';
import INITIAL_STATE from '../reducers/initialState';

let wrapper;

describe('GitHubProfileViewer', () => {
    beforeEach(() => {
        wrapper = shallow(<GitHubProfileViewer />);
    });

    it('renders without crashing', () => {
        const middlewares = [reduxThunk];
        const mockStore = configureStore(middlewares);
        const store = mockStore(INITIAL_STATE);
        const div = document.createElement('div');
        ReactDOM.render(<Provider store={store}><GitHubProfileViewer /></Provider>, div);
    });

    it('matches the snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders only one GitHubProfileViewer', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('renders an h3 element', () => {
        expect(wrapper.find('h3').length).toEqual(1);
    });

    it('renders a Form component', () => {
        expect(wrapper.find('Connect(Form)').length).toEqual(1);
    });

    it('renders a CardList component', () => {
        expect(wrapper.find('Connect(CardList)').length).toEqual(1);
    });
});
