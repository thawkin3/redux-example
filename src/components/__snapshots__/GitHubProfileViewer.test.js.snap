// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubProfileViewer matches the snapshot 1`] = `
<div>
  <h3>
    Github Profile Viewer
  </h3>
  <Connect(Form) />
  <Connect(CardList) />
</div>
`;
